fdbserver/tester.actor.cpp

```diff
@@ -1781,6 +1791,55 @@ ACTOR Future<Void> initializeSimConfig(Database db) {
 	}
 }
 
+void encryptionAtRestPlaintextMarkerCheck() {
+	if (!g_simulator->dataAtRestPlaintextMarker.present()) {
+		// Encryption at-rest was not enabled, do nothing
+		return;
+	}
+
+	ASSERT(g_simulator->isSimulated());
+
+	namespace fs = boost::filesystem;
+
+	printf("EncryptionAtRestPlaintextMarkerCheckStart\n");
+	TraceEvent("EncryptionAtRestPlaintextMarkerCheckStart");
+	fs::path p("simfdb/");
+	fs::recursive_directory_iterator end;
+	int scanned = 0;
+	bool success = true;
+	// Enumerate all files in the "simfdb/" folder and look for "marker" string
+	for (fs::recursive_directory_iterator itr(p); itr != end; ++itr) {
+		if (boost::filesystem::is_regular_file(itr->path())) {
+			std::ifstream f(itr->path().string().c_str());
+			if (f) {
+				std::string buf;
+				int count = 0;
+				while (std::getline(f, buf)) {
+					// SOMEDAY: using 'std::boyer_moore_horspool_searcher' would significantly improve search
```

Comment:
Is there a reason we cant use it in the current implementation?

Reply:
C++17 and above supports this method; current clang/gcc version has no support for it.